lib/event_listeners.js

```diff
@@ -260,6 +260,14 @@ AWS.EventListeners = {
       resp.httpResponse.body = new AWS.util.Buffer('');
       resp.httpResponse.buffers = [];
       resp.httpResponse.numBytes = 0;
+      var dateHeader = headers.date || headers.Date;
+      if (dateHeader) {
+        var serverTime = Date.parse(dateHeader);
+        if (resp.request.service.config.correctClockSkew
+            && AWS.util.isClockSkewed(serverTime)) {
```

Comment:
I know it's unlikely that an AWS response will be served from a cache, but I think it would make sense to take account of the Age header here. This should default to 0, and the argument passed to `AWS.util.isClockSkewed` should be `serverTime + date * 1000`.